x/emissions/types/validations.go

```diff
@@ -1,18 +1,764 @@
 package types
 
 import (
+	"encoding/hex"
+
 	"cosmossdk.io/errors"
+	cosmosMath "cosmossdk.io/math"
 	alloraMath "github.com/allora-network/allora-chain/math"
+	"github.com/cometbft/cometbft/crypto/secp256k1"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
+/// EXTERNAL TYPE VALIDATIONS
```

Comment:
Why throw all the validations (dec, int, all types, etc) here?

Reply:
because I was finding it really hard to find them in the various disparate files when I was looking for them

if you're already in the codebase somewhere e.g. in worker_payload.go and you hit the "jump to code" command it'll automatically open the validate function in the right file and that's fine and easy

but if you're somewhere else thinking "I want to look at the validation for x type" you have to trawl through a crowded types folder full of different filenames, instead of just thinking "i wanted to look at the validations" go straight to validations.go

also if you want to look at multiple validations at the same time easier to have them in one spot then have to look for each separately

Reply:
I softly don't support files like this (e.g. this and `keeper.go` both poorly formed files). A compromise here is to split this up into many files within a validations package

Reply:
I see where you're coming from @kpeluso with the "long-ass files" argument.

So long as these functions stay as methods of the structs that they process, they have to live in a types package, I can see us doing what you describe (a separate validations package) but then those need to be exported functions that take the things they're validating as args, as opposed to being methods on structs

Reply:
I can implement that, if you really want in this PR

Reply:
it's fine. tbh my suggestion might've been worse than just leaving them as they were